### PR TITLE
Adding dpad emulation to left stick when psx not in analog mode

### PR DESCRIPTION
**Pull request by Dybuk**
This change will add dpad emulation to the left analog stick for PS1 similar the other cores. For PSX we only do this when the analog mode is disabled.

Give me a couple of days to test this.

I might also want to consider adding an option to disable the key combos. Games like Wing Command 3 use a lot of their own combos and these blocked. 

https://www.wcnews.com/controls.shtml#wc3ps

### CODE REVIEW

**Comment by sevdestruct** [comment]
looks like you missed implementing this for standardGamepad as well. Should be right below the extended set

**Comment by sevdestruct** [comment]
Thanks for the heads up on WC3, can you add that note and link to #679 to keep in mind the later iterations of MFi+ Mode

**Comment by Dybuk** [comment]
By the standard gamepad do you mean the non-extended version of the mfi controller? I thought that didn't have analog controls. Maybe I got that wrong. I'll take a look.

**Comment by jasarien** [comment]
@Dybuk you’re right, only extended controllers have analog sticks. Standard doesn’t have them. 

It also seems like Apple are putting less emphasis on the standard profile — it’s not even mentioned in their programming guide for the game controller framework  and I think nearly all MFi game controllers still being sold today are extended profile.

**Comment by Dybuk** [comment]
I guess I could add it to the standard gamepad incase in the future we have an analog stick in the overlay controller.

**Comment by sevdestruct** [comment]
Right. Not sure why I mentioned. Hah, my bad. standard has no sticks. Ignore what I said.

**Comment by sevdestruct** [comment]
Testrun complete, works for me.

File: Mednafen/MednafenGameCore.mm [hunk 1]
```diff
@@ -1363,7 +1365,7 @@ - (void)didReleaseWSButton:(PVWSButton)button forPlayer:(NSInteger)player;
     inputBuffer[player][0] &= ~(1 << WSMap[button]);
 }
 
-- (NSInteger)controllerValueForButtonID:(unsigned)buttonID forPlayer:(NSInteger)player {
+- (NSInteger)controllerValueForButtonID:(unsigned)buttonID forPlayer:(NSInteger)player:(bool)analogMode {
```
Comment (JoeMatt) [inline Mednafen/MednafenGameCore.mm]: This is missing a parameter label, I'm not even sure how t his

File: Mednafen/MednafenGameCore.mm [hunk 2]
```diff
@@ -2038,13 +2040,31 @@ - (NSInteger)PSXcontrollerValueForButtonID:(unsigned)buttonID forController:(GCC
         bool modifiersPressed = modifier1Pressed && modifier2Pressed;
         switch (buttonID) {
             case PVPSXButtonUp:
-                return [[dpad up] isPressed];
+                if (analogMode) {
+                    return [[dpad up] isPressed];
+                }
+                else {
+                    return [[dpad up] isPressed]?:[[[pad leftThumbstick] up] isPressed];
```
Comment (JoeMatt) [inline Mednafen/MednafenGameCore.mm]: Why the use of the ternary here? A short circuited '||' would do the same thing and more readable.
Really can be 1 line return `[[dpad up] isPressed] || (!analogMode && [[[pad leftThumbstick] up] isPressed]) for all these cases same applies

**Pull request merged**